### DISCUSSION TITLE
Edit all BMS settings via BLE / ESPhome

### DISCUSSION THREAD

**Discussion by dominikfe** [syssi/esphome-jk-bms#127] [Ideas]
Hi!
Would it be feasible to add possibility to change  e.g. Balancing Start Voltage or Balance Trigger Voltage as Features?
Thx

**Comment by syssi** [comment]
Sure! I will prepare a first draft later the week. I would be happy if you provide some testing.

**Comment by dominikfe** [comment]
Absolutely! I will test it.
Would be nice to be able to set all configurable parameters.

**Reply by syssi** [reply]
Which BMS model do you own? Exposing all settings is a lot of work. I will focus on the most important ones first.

**Comment by dominikfe** [comment]
I have a JK BMS
Thank you so much!

**Reply by syssi** [reply]
Which model? I'm asking because of the protocol (JK02 vs. JK04).

**Reply by dominikfe** [reply]
Sorry, did not mention my reply wasn't sent. I have a JK02 config.

**Reply by syssi** [reply]
I've already made some progress here: https://github.com/syssi/esphome-jk-bms/pull/128

But please **don't** give it a try yet. The alignment/precision of the values doesn't fit yet.

**Reply by syssi** [reply]
Please try release 1.1.0: https://github.com/syssi/esphome-jk-bms/releases/tag/1.1.0

The most important settings can be controlled now.

**Comment by dominikfe** [comment]
Do I have to set any value upfront somewhere or do I just add the content to my yaml file and compile/upload?

**Reply by syssi** [reply]
Just add the new entities to your yaml and compile+upload. If you use Home Assistant you will see new entities populated with the current settings which can be edited easily.

**Comment by dominikfe** [comment]
It works! No issues so far!

**Comment by dominikfe** [comment]
It’s a JK02 - the latest 200A version with 2A balancing current
